### ISSUE TITLE
Desktop dll hardcodes other libraries exports

### ISSUE THREAD

**Issue by asklar** [microsoft/react-native-windows#6001]
This is not only fragile because it's not guaranteed to be stable across compiler releases, but also whenever we take a new drop of a dependency (like folly), these exports are bound to change.

See: 
- https://github.com/microsoft/react-native-windows/blob/c69af5e9826330fe98edea188c7448a82a2d5076/vnext/Desktop.DLL/react-native-win32.x64.def
- https://github.com/microsoft/react-native-windows/blob/0217d07d84c09507c1219048d62d07d3e5566796/vnext/Desktop.DLL/react-native-win32.x86.def

**Comment by chrisglein** [comment]
Sounds like this was a pain point for taking a Folly drop and made backporting that fix to 0.61 and 0.60. So would be good cleanup to limit that pain in the future.

**Comment by chrisglein** [comment]
@vmoroz I know there have been discussions in the past about refactoring this boundary. What's the current thinking on this? Do we have a plan and are waiting on the right time/priority to execute? (certainly this spring/summer has been a hard time to fit in stretch goals)
